cs/src/core/Bond.cs

```diff
@@ -3,7 +3,27 @@
 
 namespace Bond
 {
-    using System.Collections.Generic;
+    using System;
+    using System.Collections.Generic;
+
+    public partial class GUID
+    {
+        public static implicit operator Guid(GUID bondGuid)
+        {
+            return new Guid((int)bondGuid.Data1, (short)bondGuid.Data2, (short)bondGuid.Data3, BitConverter.GetBytes((long)bondGuid.Data4));

```

Comment:
Can we do the bit manipulation on `Data4` ourselves to avoid the `byte[]` allocation from `BitConverter.GetByte`?

Something like this (though, watch out for endianess issues; I might have the conversion backward):
```
(short)bondGuid.Data3,
(byte)bondGuid.Data4,
(byte)bondGuid.Data4 >> 8,
...);
```